cwl/src/main/scala/cwl/CwlExpressionCommandPart.scala

```diff
@@ -13,46 +13,46 @@ import wom.values._
 import wom.{CommandPart, InstantiatedCommand}
 
 import scala.util.Try
+import cats.syntax.either._
+
 import scala.language.postfixOps
 
 case class CwlExpressionCommandPart(expr: Expression) extends CommandPart {
   override def instantiate(inputsMap: Map[LocalName, WomValue],
                            functions: IoFunctionSet,
                            valueMapper: (WomValue) => WomValue,
-                           runtimeEnvironment: RuntimeEnvironment): ErrorOr[InstantiatedCommand] = {
-    val stringKeyMap = inputsMap.map { case (LocalName(localName), value) => localName -> value }
-
-    val pc =
-      ParameterContext(
-        runtime = runtimeEnvironment.cwlMap
-      ).withInputs(stringKeyMap, functions)
+                           runtimeEnvironment: RuntimeEnvironment): ErrorOr[InstantiatedCommand] =
 
-    expr.fold(EvaluateExpression).apply(pc).toErrorOr.map(v => InstantiatedCommand(v.valueString))
-  }
+        ParameterContext().
+          addLocalInputs(inputsMap).
+          map(_.setRuntime(runtimeEnvironment)).
+          flatMap(pc => expr.fold(EvaluateExpression).apply(pc).toEither.leftMap(e => NonEmptyList.one(e.getMessage))).
+          map(_.valueString).
+          map(s => InstantiatedCommand(s)).
```

Comment:
`map(InstantantiatedCommand.apply)`

Reply:
best I can do is `map(InstantiatedCommand.apply(_))` it has a second defaulted argument